pypowerwall/cloud/pypowerwall_cloud.py

```diff
@@ -0,0 +1,997 @@
+from pypowerwall.cloud.stubs import *
+from pypowerwall.pypowerwall_base import PyPowerwallBase
+
+log = logging.getLogger(__name__)
+
+AUTHFILE = ".pypowerwall.auth"  # Stores auth session information
+SITEFILE = ".pypowerwall.site"  # Stores site id
+COUNTER_MAX = 64  # Max counter value for SITE_DATA API
+SITE_CONFIG_TTL = 59  # Site config cache TTL in seconds
+
+
+def set_debug(debug=False, quiet=False, color=True):
+    logging.basicConfig(format='%(levelname)s: %(message)s')
+    if not quiet:
+        log.setLevel(logging.INFO)
+        if color:
+            logging.basicConfig(format='\x1b[31;1m%(levelname)s: %(message)s\x1b[0m')
+        if debug:
+            log.setLevel(logging.DEBUG)
+    else:
+        log.setLevel(logging.NOTSET)
+
+
+def lookup(data, keylist):
+    """
+    Lookup a value in a nested dictionary or return None if not found.
+        data - nested dictionary
+        keylist - list of keys to traverse
+    """
+    for key in keylist:
+        if key in data:
+            data = data[key]
+        else:
+            return None
+    return data
+
+
+# noinspection PyMethodMayBeStatic
+class PyPowerwallCloud(PyPowerwallBase):
+    def __init__(self, email: Optional[str], pwcacheexpire: int = 5, timeout: int = 5, siteid: Optional[int] = None,
+                 authpath: str = ""):
+        super().__init__(email)
+        self.site = None
+        self.tesla = None
+        self.apilock = {}  # holds lock flag for pending cloud api requests
+        self.pwcachetime = {}  # holds the cached data timestamps for api
+        self.pwcache = {}  # holds the cached data for api
+        self.pwcacheexpire = pwcacheexpire  # seconds to expire cache
+        self.siteindex = 0  # site index to use
+        self.siteid = siteid  # site id to use
+        self.counter = 0  # counter for SITE_DATA API
+        self.authpath = os.path.expanduser(authpath)  # path to cloud auth and site files
+        self.timeout = timeout
+        self.authfile = os.path.join(self.authpath, AUTHFILE)
+        self.sitefile = os.path.join(self.authpath, SITEFILE)
+        self.poll_api_map = self.init_poll_api_map()
+
+        if self.siteid is None:
+            # Check for site file
+            if os.path.exists(self.sitefile):
+                with open(self.sitefile) as file:
+                    try:
+                        self.siteid = int(file.read())
+                    except Exception as exc:
+                        log.debug(f"Unable to determine siteid from sitefile, using ID '0' instead: {exc}")
+                        self.siteid = 0
+            else:
+                self.siteindex = 0
+        log.debug(f" -- cloud: Using site {self.siteid} for {self.email}")
+
+        # Check for auth file
+        if not os.path.exists(self.authfile):
+            msg = f"Missing auth file {self.authfile} - run setup"
+            log.warning(msg)
+            raise PyPowerwallCloudNoTeslaAuthFile(msg)
```

Comment:
This is the problem code.  In `__main__.py` when users call for `setup` it will initialize this class. This conditional will error because the user hasn't run setup before. Unfortunately, the only way to run setup to create the file is to init this class to get to self.setup().

We should move this to connect()?

Reply:
I will look into it.

Reply:
@jasonacox oddly enough, the `connect()` method already had a check for the auth file. It makes sense to remove it from init, I am still looking for consequences of doing it, but it looks viable.

Reply:
Looks good to remove it from the init method.